### PR TITLE
chore: update code to fix sonar cloud issue

### LINKED ISSUE #1403: [REFACTOR] Reduce return value should be used
https://sonarcloud.io/project/issues?pullRequest=1395&open=AZIFOp3Raj5kJiILYjG2&id=graasp_graasp

This code has several issues:
- we do not use the return value from the `reduce`
- we use instead the constant that is modified ... which is really not a good practice.

### PR DESCRIPTION
**Pull request by spaenleh**
@Defferrard this should fix your sonarCloud issues. It should also improve the code's reliability.

fix #1403

### CODE REVIEW

File: src/services/itemMembership/repository.ts [hunk 1]
```diff
@@ -703,24 +703,25 @@ export class ItemMembershipRepository extends MutableRepository<
...
-      const { accountId, itemPath, permission } = row;
-      if (itemPath !== item.path) {
-        chngs.inserts.push({
-          accountId,
-          itemPath: itemIdAsPath,
-          permission,
-          creatorId: account.id,
-        });
-      }
+    const changes = rows.reduce<ResultMoveHousekeeping>(
```
Comment (ReidyT) [inline src/services/itemMembership/repository.ts:706]:
```suggestion
    return rows.reduce<ResultMoveHousekeeping>(
```

File: src/services/itemMembership/repository.ts [hunk 2]
```diff
@@ -703,24 +703,25 @@ export class ItemMembershipRepository extends MutableRepository<
...
+      (chngs, row) => {
```
Comment (ReidyT) [inline src/services/itemMembership/repository.ts:707]:
```suggestion
      (changes, row) => {
```

File: src/services/itemMembership/repository.ts [hunk 3]
```diff
@@ -776,48 +777,49 @@ export class ItemMembershipRepository extends MutableRepository<
...
+        if (action === 2 && action2IgnoreInherited) {
+          return chngs;
+        }
 
-      // permission worse or equal to inherited one at "destination"
-      if (action === 2 && PermissionLevelCompare.lte(permission, inherited)) {
-        chngs.deletes.push({ accountId, itemPath });
-      }
+        const permission = p;
+        const inherited = ip;
```
Comment (ReidyT) [inline src/services/itemMembership/repository.ts:799]: Why it is not possible to use the `permission` and `inherited` directly from the `row`?
```suggestion
```

File: src/services/itemMembership/repository.ts [hunk 4]
```diff
@@ -776,48 +777,49 @@ export class ItemMembershipRepository extends MutableRepository<
...
+    const changes = rows.reduce<ResultMoveHousekeeping>(
+      (chngs, row) => {
+        const {
           accountId,
           itemPath,
-          permission,
-          creatorId: accountId,
-        });
-      }
+          permission: p,
```
Comment (ReidyT) [inline src/services/itemMembership/repository.ts:785]:
```suggestion
          permission,
```

File: src/services/itemMembership/repository.ts [hunk 5]
```diff
@@ -776,48 +777,49 @@ export class ItemMembershipRepository extends MutableRepository<
...
+          action,
+          inherited: ip,
```
Comment (ReidyT) [inline src/services/itemMembership/repository.ts:787]:
```suggestion
          inherited,
```

File: src/services/itemMembership/repository.ts [hunk 6]
```diff
@@ -776,48 +777,49 @@ export class ItemMembershipRepository extends MutableRepository<
...
-      }
-
-      const permission = p;
-      const inherited = ip;
-
-      // permission (inherited) at the "origin" better than inherited one at "destination"
-      if (action === 1 && PermissionLevelCompare.gt(permission, inherited)) {
-        chngs.inserts.push({
+    const changes = rows.reduce<ResultMoveHousekeeping>(
+      (chngs, row) => {
```
Comment (ReidyT) [inline src/services/itemMembership/repository.ts:781]:
```suggestion
   return rows.reduce<ResultMoveHousekeeping>(
      (changes, row) => {
```

File: src/services/itemMembership/repository.ts [hunk 7]
```diff
@@ -776,48 +777,49 @@ export class ItemMembershipRepository extends MutableRepository<
...
-          creatorId: accountId,
-        });
-      }
+          permission: p,
+          action,
+          inherited: ip,
+          action2IgnoreInherited,
+        } = row;
+
+        if (action === 0) {
```
Comment (ReidyT) [inline src/services/itemMembership/repository.ts:791]: What action 0, 1, or 2 means? Should we use an enum here?

**Review by ReidyT — CHANGES_REQUESTED** [review]
Thanks for this PR, LGTM! I just let some suggestions to avoid using misspelling variables (like `chnges`) and to avoid using magic numbers. Let me know if you disagree 😆.

File: src/services/itemMembership/repository.ts (hunk 3 above)
Comment (spaenleh) [reply src/services/itemMembership/repository.ts:799]: I have no f***!ng idea 🤷

**Comment by spaenleh** [comment]
@ReidyT I have made changes according to your comment. Let me know If you think this is clearer... I honestly do not really understand what this actually does, so i tried to improve the code as best as I could.

**Review by ReidyT — APPROVED** [review]
Thanks for your changes, LGTM 🚀

**Comment by ReidyT** [comment]
> @ReidyT I have made changes according to your comment. Let me know If you think this is clearer... I honestly do not really understand what this actually does, so i tried to improve the code as best as I could.

Yes it's clearer, I know it was small issues, but thanks for your changes 👍

**Pull request merged**